testdata/project-v4-with-plugins/internal/controller/memcached_controller.go

```diff
@@ -117,37 +118,22 @@ func (r *MemcachedReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		}
 	}
 
-	// Let's add a finalizer. Then, we can define some operations which should
-	// occur before the custom resource is deleted.
-	// More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/finalizers
-	if !controllerutil.ContainsFinalizer(memcached, memcachedFinalizer) {
-		log.Info("Adding Finalizer for Memcached")
-		if ok := controllerutil.AddFinalizer(memcached, memcachedFinalizer); !ok {
-			log.Error(err, "Failed to add finalizer into the custom resource")
-			return ctrl.Result{Requeue: true}, nil
-		}
-
-		if err = r.Update(ctx, memcached); err != nil {
-			log.Error(err, "Failed to update custom resource to add finalizer")
-			return ctrl.Result{}, err
-		}
-	}
-
 	// Check if the Memcached instance is marked to be deleted, which is
 	// indicated by the deletion timestamp being set.
-	isMemcachedMarkedToBeDeleted := memcached.GetDeletionTimestamp() != nil
```

Comment:
Hi @mateusoliveira43 

IMO: We need to be very surgical about the changes. 
The 

```
ptr.To(true),
RunAsUser:                ptr.To(int64(1001)),
AllowPrivilegeEscalation: ptr.To(false),
```

is a good enhancement

And we might can change the place of the finalizer to create an error and log it as return it
However, I do not see reasons for the others changes.

Reply:
update PR